core/src/main/java/org/apache/lucene/queryparser/classic/QueryParserSettings.java

```diff
@@ -19,64 +19,72 @@
 
 package org.apache.lucene.queryparser.classic;
 
-import com.carrotsearch.hppc.ObjectFloatHashMap;
 import org.apache.lucene.analysis.Analyzer;
-import org.apache.lucene.search.FuzzyQuery;
 import org.apache.lucene.search.MultiTermQuery;
-import org.apache.lucene.util.automaton.Operations;
 import org.elasticsearch.common.unit.Fuzziness;
 import org.joda.time.DateTimeZone;
 
-import java.util.List;
 import java.util.Locale;
+import java.util.Map;
 
 /**
- *
+ * Encapsulates settings that affect query_string parsing via {@link MapperQueryParser}
  */
 public class QueryParserSettings {
```

Comment:
I gave a try to this in the beginning, also at having settings as part of QueryStringQueryBuilder. The issue I encountered is that there's default values handling, some defaults that come from settings, small things that get modified once we go and create the actual query (logic is now in toQuery). The current solution does duplicate quite a few settings, but it feels a bit cleaner, given that QueryParserSettings holds the actual values that will be used for the query, while the query buidler holds the user values that might change once we go and execute the query (aka call toQuery). Should I look into it again? WDYT?